ocaml/seahash/libseahash/seahash.c

```diff
@@ -0,0 +1,144 @@
+#include <stdint.h>
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+
+/*
+	This code is a C implementation of the SeaHash algorithm (see: http://ticki.github.io/blog/seahash-explained/)
+	The author also provides a reference implemenation written in Rust (which follows the MIT license). 
+*/
+
+#define BUFF_SIZE (61440)
+
+uint64_t A, B, C, D;
```

Comment:
It is usually best to avoid global variables, especially when dealing with threaded code. Even though in its current state OCaml is single threaded, that might change in the future, and we can also explicitly drop the runtime lock if we'd be using bigstrings that do not move.
It would be best to put these into a struct.